spicy/toolchain/src/compiler/parser/parser.yy

```diff
@@ -888,12 +907,11 @@ ctor          : CADDRESS                         { $$ = hilti::ctor::Address(hil
 
               | const_real                       { $$ = hilti::ctor::Real($1, __loc__); }
               | CUINTEGER                        { $$ = hilti::ctor::UnsignedInteger($1, 64, __loc__); }
```

Comment:
I am confused why we check some uses of `CUINTEGER`, but not others (e.g., this one here). Shouldn't we range-check _all uses_ of `CUINTEGER`?

Here and elsewhere, similarly in the hilti parser.

Reply:
We only need to range check cases where the `CUINTEGER` then turns into signed value. If it's used as an actual unsigned going forward, it'll always be in range (because we're already storing it in an uint64 anyways when we arrive in parser.yy)